backend/src/hatchling/metadata/core.py

```diff
@@ -292,40 +312,46 @@ class CoreMetadata:
     https://peps.python.org/pep-0621/
     """
 
-    def __init__(self, root, config, hatch_metadata, context):
+    def __init__(
+        self,
+        root: Path | str,
+        config: dict[str, Any],
+        hatch_metadata: "HatchMetadataSettings",
+        context: Context,
+    ) -> None:
         self.root = root
         self.config = config
         self.hatch_metadata = hatch_metadata
         self.context = context
 
-        self._raw_name = None
-        self._name = None
-        self._version = None
-        self._description = None
-        self._readme = None
-        self._readme_content_type = None
-        self._readme_path = None
-        self._requires_python = None
-        self._python_constraint = None
-        self._license = None
-        self._license_expression = None
-        self._license_files = None
-        self._authors = None
-        self._authors_data = None
-        self._maintainers = None
-        self._maintainers_data = None
-        self._keywords = None
-        self._classifiers = None
+        self._raw_name: str | None = None
+        self._name: str | None = None
+        self._version: str | None = None
+        self._description: str | None = None
+        self._readme: str | None = None
+        self._readme_content_type: str | None = None  # TODO: make into enum?
```

Comment:
There are only three readme content types, keeping them in a str enum would make it a bit more obvious what their allowed values are.

However the types are inferred from the file type anyway, and they're unlikely to change, so this wouldn't really have many benefits.

Reply:
Yeah let's keep it as-is